### ISSUE TITLE
[mwc-select] menu is cut at the top when opens in top position

### ISSUE THREAD

**Issue by yinonov** [material-components/material-components-web-components#2409]
**Describe the bug**
select inside a dialog is being cut at the top when menu opens at the top (due to lack of block space)

**To Reproduce**
1. open select inside a "short" dialog window
2. make sure the top is more spacious so that menu opens at the top

**Expected behavior**
native `select` would open as fixed and wont let user interact (scroll) until click outside and select is closed
`fixedMenuPosition` mixed with scrim might achieve the same expected result

**Screenshots**
the video shows a ״vexing״ case but in practice this issue becomes quite annoying.

https://user-images.githubusercontent.com/15140652/118017259-11648f00-b35f-11eb-9d9e-d769be3b5510.mov

**Comment by e111077** [comment]
Heya does fixedMenuPosition work? It's a bit unclear from your report. Also would it be possible to come up with a live repro? Such as on jsbin or the https://lit.dev/playground

**Comment by yinonov** [comment]
the desired result is not like with `fixedMenuPosition`. the menu should follow its anchor when scrolled.

For some reason lit's playground behaves differently from codepen (on my machine at least) and opens the menu of the basic select at the bottom of anchor while in codepen it is opened at the top and gets cut partially at its top
###### lit's playground
![image](https://user-images.githubusercontent.com/15140652/119259173-b4c26900-bbd5-11eb-97da-c879195f5e49.png)
###### codepen
![image](https://user-images.githubusercontent.com/15140652/119259162-a411f300-bbd5-11eb-9b7e-186959b0372a.png)

verify yourself and click the 1st select without scrolling (on codepen) -

- [codepen](https://codepen.io/yinonov/pen/zYZZbWa)

- [lit playground](https://lit.dev/playground/#project=W3sibmFtZSI6ImluZGV4Lmh0bWwiLCJjb250ZW50IjoiPCFET0NUWVBFIGh0bWw-XG48aGVhZD5cbiAgPHNjcmlwdCB0eXBlPVwibW9kdWxlXCIgc3JjPVwiaHR0cHM6Ly9jZG4uc2t5cGFjay5kZXYvQG1hdGVyaWFsL213Yy1kaWFsb2dAMC4yMi4wLWNhbmFyeS4wMTJjOGRjMC4wXCI-PC9zY3JpcHQ-XG4gIDxzY3JpcHQgdHlwZT1cIm1vZHVsZVwiIHNyYz1cImh0dHBzOi8vY2RuLnNreXBhY2suZGV2L0BtYXRlcmlhbC9td2Mtc2VsZWN0QDAuMjIuMC1jYW5hcnkuMDEyYzhkYzAuMFwiPjwvc2NyaXB0PlxuICA8c2NyaXB0IHR5cGU9XCJtb2R1bGVcIiBzcmM9XCJodHRwczovL2Nkbi5za3lwYWNrLmRldi9AbWF0ZXJpYWwvbXdjLWxpc3RAMC4yMi4wLWNhbmFyeS4wMTJjOGRjMC4wXCI-PC9zY3JpcHQ-XG4gIDxzdHlsZT5cbiAgICBtd2MtZGlhbG9nIHtcbiAgICAgIC0tbWRjLWRpYWxvZy1tYXgtaGVpZ2h0OiAyMDBweDtcbiAgICB9XG4gIDwvc3R5bGU-XG48L2hlYWQ-XG48Ym9keT5cbiAgPG13Yy1kaWFsb2cgb3Blbj5cbiAgICA8ZGw-XG4gICAgICA8ZHQ-QmFzaWMgc2VsZWN0PC9kdD5cbiAgICAgIDxkZD5cbiAgICAgICAgPHA-bWVudSBvcGVucyBhdCB0b3Agb2YgYW5jaG9yIGlzIHBhcnRpYWxseSBvZmZzZXQgJiBoaWRkZW48L3A-XG4gICAgICAgIDxtd2Mtc2VsZWN0PlxuICAgICAgICAgIDxtd2MtbGlzdC1pdGVtPml0ZW0gMTwvbXdjLWxpc3QtaXRlbT5cbiAgICAgICAgICA8bXdjLWxpc3QtaXRlbT5pdGVtPC9td2MtbGlzdC1pdGVtPlxuICAgICAgICAgIDxtd2MtbGlzdC1pdGVtPml0ZW08L213Yy1saXN0LWl0ZW0-XG4gICAgICAgICAgPG13Yy1saXN0LWl0ZW0-aXRlbTwvbXdjLWxpc3QtaXRlbT5cbiAgICAgICAgICA8bXdjLWxpc3QtaXRlbT5pdGVtPC9td2MtbGlzdC1pdGVtPlxuICAgICAgICA8L213Yy1zZWxlY3Q-XG4gICAgICA8L2RkPlxuICAgICAgPGR0Pk5hdGl2ZTwvZHQ-XG4gICAgICA8ZGQ-XG4gICAgICAgIDxwPm9uY2Ugb3BlbmVkLCBuYXRpdmUgc2VsZWN0IDxzdHJvbmc-cmVxdWlyZSBjbGljayBvdXRzaWRlIGFuZCBjbG9zaW5nIHRoZSBtZW51PC9zdHJvbmc-LCBpbiBvcmRlciB0byBjb250aW51ZSBzY3JvbGxpbmcuPGJyPiDinJTvuI9JIHRoaW5rIGl0J3MgdGhlIHdheSB0byBnbzwvcD5cbiAgICAgICAgPHNlbGVjdCBuYW1lPVwiY2Fyc1wiIGlkPVwiY2Fyc1wiPlxuICAgICAgICAgIDxvcHRpb24gdmFsdWU9XCJ2b2x2b1wiPlZvbHZvPC9vcHRpb24-XG4gICAgICAgICAgPG9wdGlvbiB2YWx1ZT1cInNhYWJcIj5TYWFiPC9vcHRpb24-XG4gICAgICAgICAgPG9wdGlvbiB2YWx1ZT1cIm9wZWxcIj5PcGVsPC9vcHRpb24-XG4gICAgICAgICAgPG9wdGlvbiB2YWx1ZT1cImF1ZGlcIj5BdWRpPC9vcHRpb24-XG4gICAgICAgIDwvc2VsZWN0PlxuICAgICAgPC9kZD5cbiAgICAgIDxkdD5GaXhlZE1lbnVQb3NpdGlvbjwvZHQ-XG4gICAgICA8ZGQ-XG4gICAgICAgIDxwPnNldHRpbmc8a2JkPmZpeGVkTWVudVBvc2l0aW9uPC9rYmQ-IG9uIHNlbGVjdCB3aWxsIGtlZXAgbWVudSBvcGVuZWQgYW5kIGF0IHRoZSBzYW1lIHBsYWNlIGV2ZW4gd2hlbiBzY3JvbGxpbmcgYXdheSBpdHMgYW5jaG9yIChvcGVuIHRoZSBzZWxlY3QgYW5kIHNjcm9sbCk8L3A-XG4gICAgICAgIDxtd2Mtc2VsZWN0IGZpeGVkTWVudVBvc2l0aW9uPlxuICAgICAgICAgIDxtd2MtbGlzdC1pdGVtPml0ZW0gMTwvbXdjLWxpc3QtaXRlbT5cbiAgICAgICAgICA8bXdjLWxpc3QtaXRlbT5pdGVtPC9td2MtbGlzdC1pdGVtPlxuICAgICAgICAgIDxtd2MtbGlzdC1pdGVtPml0ZW08L213Yy1saXN0LWl0ZW0-XG4gICAgICAgICAgPG13Yy1saXN0LWl0ZW0-aXRlbTwvbXdjLWxpc3QtaXRlbT5cbiAgICAgICAgICA8bXdjLWxpc3QtaXRlbT5pdGVtPC9td2MtbGlzdC1pdGVtPlxuICAgICAgICA8L213Yy1zZWxlY3Q-XG4gICAgICA8L2RkPlxuICAgIDwvZGw-XG4gICAgPHA-XG4gICAgICBMb3JlbSBpcHN1bSBkb2xvciBzaXQgYW1ldCBjb25zZWN0ZXR1ciBhZGlwaXNpY2luZyBlbGl0LiBJbmNpZHVudCB2aXRhZSBlc3NlIGRlbGVuaXRpIHRvdGFtIHF1YW0gZXJyb3IgaXBzYW0gYW1ldCBkb2xvcnVtIGl1c3RvLCBhY2N1c2FtdXMgZXggZG9sb3JlbSwgbmFtIHF1b2QgbmloaWwgZHVjaW11cyEgTm9uIGlsbHVtIG1haW9yZXMgcHJvdmlkZW50LlxuICAgICAgTG9yZW0gaXBzdW0gZG9sb3IsIHNpdCBhbWV0IGNvbnNlY3RldHVyIGFkaXBpc2ljaW5nIGVsaXQuIExhYm9ydW0gb2RpbyBxdW8gcGVyZmVyZW5kaXMgbW9sbGl0aWEgbWluaW1hIHVsbGFtIG5vYmlzIHF1b2Qgc2FlcGUgcHJhZXNlbnRpdW0sIHRlbmV0dXIgZGlzdGluY3RpbyBxdWlkZW0gZXVtLCBudW1xdWFtIHZvbHVwdGF0dW0gcmVwcmVoZW5kZXJpdCBleGVyY2l0YXRpb25lbSEgTnVtcXVhbSwgY29ycnVwdGkgbW9sZXN0aWFzIVxuICAgICAgTG9yZW0gaXBzdW0gZG9sb3Igc2l0IGFtZXQgY29uc2VjdGV0dXIgYWRpcGlzaWNpbmcgZWxpdC4gSXBzYW0gbm9iaXMgYXJjaGl0ZWN0byBiZWF0YWUgdmVuaWFtIGl1cmUgcGxhY2VhdCBldW0gYWNjdXNhbXVzLCBxdW9kIGVhcnVtPyBTaXQgcHJvdmlkZW50IGRpc3RpbmN0aW8gZnVnaXQsIG9iY2FlY2F0aSBkb2xvcnVtIGlsbHVtIGV4cGxpY2FibyB0b3RhbSEgRnVnaWF0LCBudWxsYT9cbiAgICAgIExvcmVtIGlwc3VtIGRvbG9yIHNpdCBhbWV0IGNvbnNlY3RldHVyIGFkaXBpc2ljaW5nIGVsaXQuIEluY2lkdW50IHZpdGFlIGVzc2UgZGVsZW5pdGkgdG90YW0gcXVhbSBlcnJvciBpcHNhbSBhbWV0IGRvbG9ydW0gaXVzdG8sIGFjY3VzYW11cyBleCBkb2xvcmVtLCBuYW0gcXVvZCBuaWhpbCBkdWNpbXVzISBOb24gaWxsdW0gbWFpb3JlcyBwcm92aWRlbnQuXG4gICAgICBMb3JlbSBpcHN1bSBkb2xvciwgc2l0IGFtZXQgY29uc2VjdGV0dXIgYWRpcGlzaWNpbmcgZWxpdC4gTGFib3J1bSBvZGlvIHF1byBwZXJmZXJlbmRpcyBtb2xsaXRpYSBtaW5pbWEgdWxsYW0gbm9iaXMgcXVvZCBzYWVwZSBwcmFlc2VudGl1bSwgdGVuZXR1ciBkaXN0aW5jdGlvIHF1aWRlbSBldW0sIG51bXF1YW0gdm9sdXB0YXR1bSByZXByZWhlbmRlcml0IGV4ZXJjaXRhdGlvbmVtISBOdW1xdWFtLCBjb3JydXB0aSBtb2xlc3RpYXMhXG4gICAgICBMb3JlbSBpcHN1bSBkb2xvciBzaXQgYW1ldCBjb25zZWN0ZXR1ciBhZGlwaXNpY2luZyBlbGl0LiBJcHNhbSBub2JpcyBhcmNoaXRlY3RvIGJlYXRhZSB2ZW5pYW0gaXVyZSBwbGFjZWF0IGV1bSBhY2N1c2FtdXMsIHF1b2QgZWFydW0_IFNpdCBwcm92aWRlbnQgZGlzdGluY3RpbyBmdWdpdCwgb2JjYWVjYXRpIGRvbG9ydW0gaWxsdW0gZXhwbGljYWJvIHRvdGFtISBGdWdpYXQsIG51bGxhP1xuICAgICAgTG9yZW0gaXBzdW0gZG9sb3Igc2l0IGFtZXQgY29uc2VjdGV0dXIgYWRpcGlzaWNpbmcgZWxpdC4gSW5jaWR1bnQgdml0YWUgZXNzZSBkZWxlbml0aSB0b3RhbSBxdWFtIGVycm9yIGlwc2FtIGFtZXQgZG9sb3J1bSBpdXN0bywgYWNjdXNhbXVzIGV4IGRvbG9yZW0sIG5hbSBxdW9kIG5paGlsIGR1Y2ltdXMhIE5vbiBpbGx1bSBtYWlvcmVzIHByb3ZpZGVudC5cbiAgICAgIExvcmVtIGlwc3VtIGRvbG9yLCBzaXQgYW1ldCBjb25zZWN0ZXR1ciBhZGlwaXNpY2luZyBlbGl0LiBMYWJvcnVtIG9kaW8gcXVvIHBlcmZlcmVuZGlzIG1vbGxpdGlhIG1pbmltYSB1bGxhbSBub2JpcyBxdW9kIHNhZXBlIHByYWVzZW50aXVtLCB0ZW5ldHVyIGRpc3RpbmN0aW8gcXVpZGVtIGV1bSwgbnVtcXVhbSB2b2x1cHRhdHVtIHJlcHJlaGVuZGVyaXQgZXhlcmNpdGF0aW9uZW0hIE51bXF1YW0sIGNvcnJ1cHRpIG1vbGVzdGlhcyFcbiAgICAgIExvcmVtIGlwc3VtIGRvbG9yIHNpdCBhbWV0IGNvbnNlY3RldHVyIGFkaXBpc2ljaW5nIGVsaXQuIElwc2FtIG5vYmlzIGFyY2hpdGVjdG8gYmVhdGFlIHZlbmlhbSBpdXJlIHBsYWNlYXQgZXVtIGFjY3VzYW11cywgcXVvZCBlYXJ1bT8gU2l0IHByb3ZpZGVudCBkaXN0aW5jdGlvIGZ1Z2l0LCBvYmNhZWNhdGkgZG9sb3J1bSBpbGx1bSBleHBsaWNhYm8gdG90YW0hIEZ1Z2lhdCwgbnVsbGE_XG4gICAgICBMb3JlbSBpcHN1bSBkb2xvciBzaXQgYW1ldCBjb25zZWN0ZXR1ciBhZGlwaXNpY2luZyBlbGl0LiBJbmNpZHVudCB2aXRhZSBlc3NlIGRlbGVuaXRpIHRvdGFtIHF1YW0gZXJyb3IgaXBzYW0gYW1ldCBkb2xvcnVtIGl1c3RvLCBhY2N1c2FtdXMgZXggZG9sb3JlbSwgbmFtIHF1b2QgbmloaWwgZHVjaW11cyEgTm9uIGlsbHVtIG1haW9yZXMgcHJvdmlkZW50LlxuICAgICAgTG9yZW0gaXBzdW0gZG9sb3IsIHNpdCBhbWV0IGNvbnNlY3RldHVyIGFkaXBpc2ljaW5nIGVsaXQuIExhYm9ydW0gb2RpbyBxdW8gcGVyZmVyZW5kaXMgbW9sbGl0aWEgbWluaW1hIHVsbGFtIG5vYmlzIHF1b2Qgc2FlcGUgcHJhZXNlbnRpdW0sIHRlbmV0dXIgZGlzdGluY3RpbyBxdWlkZW0gZXVtLCBudW1xdWFtIHZvbHVwdGF0dW0gcmVwcmVoZW5kZXJpdCBleGVyY2l0YXRpb25lbSEgTnVtcXVhbSwgY29ycnVwdGkgbW9sZXN0aWFzIVxuICAgIDwvcD5cbiA8L213Yy1kaWFsb2c-XG48L2JvZHk-XG4ifV0)

**Comment by yinonov** [comment]
notice the difference of native select against mwc in reference to `fixedMenuPosition`. seems like native is fixed as well but it wont let user scroll away without closing its menu. maybe setting `fixedMenuPosition` on `mwc-select` should add same logic

**Comment by e111077** [comment]
This is due to the lack of menu hoisting. This is something we expect to tackle before 1.0.0. Current workaround is `<mwc-select fixedMenuPosition>` closing as duplicate of #832

**Comment by e111077** [comment]
Heya, yeah, the correct way to implement this is to actually render the menu at document.body. We're still trying to figure out a method to do this such that it's not lit specific and works across frameworks

**Comment by yinonov** [comment]
I'm not sure how material you will come into play but at the moment a lot of the styling of material web components is handled by css variables as such there are use-cases where theming (css variables) is applied via components as well.
```html
<my-theme dark>
  <mwc-dialog open>
    <!-- some content -->
  </mwc-dialog>
</my-theme>
```
in the above snippet, given that ***my-theme*** sets the relevant variables, and ***dialog*** moves elsewhere outside the theme component scope, it will loose styling.
not sure what is material's opinion on theming components as a whole, and what you got lined up but some sort of theming is must for consumers and the alternative would be css traditional styling, which kind of deviates from the modular pattern

**Comment by e111077** [comment]
Yeah, this is a known issue we'd have to handle as well as determining how to handle event propagation. We likely would make hoisting opt-in <mwc-select menuHoisted> or <mwc-menu hoisted>

**Comment by yinonov** [comment]
why was this closed? very much related to #2577

**Comment by e111077** [comment]
This was closed as duplicate of #832
